### ISSUE TITLE
Params decorators do not respect types (inversify-express-utils)

### ISSUE THREAD

**Issue by tspoke** [inversify/InversifyJS#979]
When using `inversify-express-utils`, the decorator @queryParam does not handle types correctly. The type of the parameter is always `string` for primitives types.

```
@httpGet("/test")
public test(@queryParam("count") count: number, @request() req, @response() res) {
  console.log(typeof count); // print 'string'
  console.log(count === 1); // print false because count is a string not a int
  return count;
}
```

## Expected Behavior

@queryParam should handle primitive types appropriately with a type-check + cast if needed.

## Current Behavior

Primitive types are always a `string`.

## Your Environment

```
    "inversify": "4.13.0"
    "inversify-express-utils": "6.0.0"
```

**Comment by tspoke** [comment]
@dcavanagh All primitives should be a good start ? (`boolean`, `string`, `number`, `integer`, `float`)

Maybe it could be interesting to extends to `array` and `enum`...

**Comment by mmis1000** [comment]
I think that problably can't be done wthiout read the type decorator generated by typescript directly(not enabled by default)

inversify have no way to know whether you want a number or boolean or string

**Comment by Jameskmonger** [comment]
All URL parameters are strings because they're in a URL, so I think this is by design.

**Comment by tspoke** [comment]
@Jameskmonger Yes, of course. Even if it's by design, casting types into the wanted format should/could be a great feature for :

• Type check on request (throwing wrong type exceptions)
• Value casting for usage without casting by hand

**Comment by grapereader** [comment]
This can introduce subtle bugs if the developer isn't aware of it.

Either the annotation should force a string type, or it should cast to the destination type. Having a string sitting around inside of what claims to be an int, or bool, or whatever... that's dangerous. It bypasses the type checker.

**Comment by descampsk** [comment]
4 years after and it is still the case. Do you think there is a possibility to fix this ?
```js
@request() req: TypeRequest
```
is casting to the right type when accessing `req.params`.
